tests/test_pytorch.py

```diff
@@ -77,20 +155,208 @@ def test_nursery_pytorch_save_entire_model():
+                              nb_classes=2)
+    model.fit(data, save_entire_model=False, nb_epochs=1)
+
+    pred = model.predict(data)
+    assert (pred.shape[0] == x.shape[0])
+    score = model.score(data)
+    assert (0 < score <= 1.0)
+
+
+def test_pytorch_predictions_multi_label_cat():
+    # This kind of model requires special training and will not be supported using the 'fit' method.
+    class multi_label_cat_model(nn.Module):
+
+        def __init__(self, num_classes, num_features):
+            super(multi_label_cat_model, self).__init__()
+
+            self.fc1 = nn.Sequential(
+                nn.Linear(num_features, 256),
+                nn.Tanh(), )
+
+            self.classifier1 = nn.Linear(256, num_classes)
+            self.classifier2 = nn.Linear(256, num_classes)
+
+        def forward(self, x):
+            out1 = self.classifier1(self.fc1(x))
+            out2 = self.classifier2(self.fc1(x))
+            return out1, out2
+
+    (x_train, y_train), (x_test, y_test) = dataset_utils.get_iris_dataset_np()
+
+    # make multi-label categorical
+    num_classes = 3
+    y_train = check_and_transform_label_format(y_train, nb_classes=num_classes)
+    y_test = check_and_transform_label_format(y_test, nb_classes=num_classes)
+    y_train = np.column_stack((y_train, y_train))
+    y_test = np.stack([y_test, y_test], axis=1)
+    test = PytorchData(x_test.astype(np.float32), y_test.astype(np.float32))
+
+    inner_model = multi_label_cat_model(num_classes, 4)
+    criterion = nn.CrossEntropyLoss()
+    optimizer = optim.Adam(inner_model.parameters(), lr=0.01)
+
+    # train model
+    train_dataset = TensorDataset(from_numpy(x_train.astype(np.float32)), from_numpy(y_train.astype(np.float32)))
+    train_loader = DataLoader(train_dataset, batch_size=100, shuffle=True)
+
+    for epoch in range(5):
+        # Train for one epoch
+        for inputs, targets in train_loader:
+            # Zero the parameter gradients
+            optimizer.zero_grad()
+
+            # Perform prediction
+            model_outputs = inner_model(inputs)
+
+            # Form the loss function
+            loss = 0
+            for i, o in enumerate(model_outputs):
+                t = targets[:, i * num_classes:(i + 1) * num_classes]
+                loss += criterion(o, t)
+
+            loss.backward()
+
+            optimizer.step()
+
+    model = PyTorchClassifier(model=inner_model,
+                              output_type=ModelOutputType.CLASSIFIER_MULTI_OUTPUT_CLASS_LOGITS,
+                              loss=criterion,
+                              optimizer=optimizer,
+                              input_shape=(24,),
+                              nb_classes=3)
+
+    pred = model.predict(test)
+    assert (pred.shape[0] == x_test.shape[0])
+
+    score = model.score(test, apply_non_linearity=expit)
+    assert (0 < score <= 1.0)
+
+
+def test_pytorch_predictions_multi_label_binary():
+    class multi_label_binary_model(nn.Module):
```

Comment:
Again, class names should normally use the CapWords convention, unless they are primarily used as a callable.